<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Creating MAMS from Seurat • rmams</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Creating MAMS from Seurat">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">rmams</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/annData_rmams_tutorial.html">Creating MAMS from AnnData</a></li>
    <li><a class="dropdown-item" href="../articles/sce_rmams_tutorial.html">Creating MAMS from SingleCellExperiment</a></li>
    <li><a class="dropdown-item" href="../articles/seurat_rmams_tutorial.html">Creating MAMS from Seurat</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Creating MAMS from Seurat</h1>
            
            <h4 data-toc-skip class="date">2024-08-28</h4>
      

      <div class="d-none name"><code>seurat_rmams_tutorial.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Matrix and Analysis Metadata Standards (MAMS) can be used to capture
the relevant information about the data matrices and annotations that
are produced during common and complex analysis workflows for
single-cell data. This use case demonstrates how MAMS can be used to
annotate matrices and annotations that are produced during the analysis
of multi-modal single-cell data containing both RNA and Antibody Derived
Tags (ADTs). We will analyze the <a href="https://support.10xgenomics.com/single-cell-gene-expression/datasets/4.0.0/SC3_v3_NextGem_SI_PBMC_CSP_1K" class="external-link">PBMC
1K dataset</a> with the Seurat package according to the <a href="https://satijalab.org/seurat/articles/multimodal_vignette.html" class="external-link">Multi-modal</a>
and <a href="https://satijalab.org/seurat/articles/weighted_nearest_neighbor_analysis.html" class="external-link">Weighted
Nearst Neighbor vignettes</a>. The analytical steps include:</p>
<ol style="list-style-type: decimal">
<li>Importing the raw/droplet and filtered/cell matrices</li>
<li>Performing quality control and further subsetting the filtered
matrix to only include high-quality cells</li>
<li>Clustering all cells with RNA only</li>
<li>Clustering all cells with ADTs only</li>
<li>Clustering all cells with both RNA and ADT</li>
<li>Clustering T-cells with RNA only</li>
<li>Clustering T-cells with ADT only</li>
<li>Clustering T-cells with both RNA and ADT</li>
</ol>
<p>This workflow demonstrates how different subsets of
cells/observations are produced due to QC/filtering or can be made based
on different biological subsets. It also produces several matrices and
annotations from different modalities. Note that the goal of this
workflow is less about producing the best clustering results and cell
type characterization. Rather it is focused on producing data/results
that is commonly generated during real-world analyses which can be
annotated with MAMS. For more information about MAMS, see the <a href="https://github.com/single-cell-mams/mams" class="external-link">GitHub
repository</a></p>
<div class="section level3">
<h3 id="step-1-creating-the-seurat-object">Step 1: Creating the Seurat object<a class="anchor" aria-label="anchor" href="#step-1-creating-the-seurat-object"></a>
</h3>
<div class="section level4">
<h4 id="importing">Importing<a class="anchor" aria-label="anchor" href="#importing"></a>
</h4>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span></code></pre></div>
<p>First, we will import raw/droplet and filtered/cell matrices:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">raw.matrices</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/Read10X.html" class="external-link">Read10X</a></span><span class="op">(</span>data.dir <span class="op">=</span> <span class="st">"../data/raw_feature_bc_matrix/"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 10X data contains more than one type and is being returned as a list containing matrices of each type.</span></span>
<span><span class="va">data.raw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">raw.matrices</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, assay <span class="op">=</span> <span class="st">"RNA"</span>, project <span class="op">=</span> <span class="st">"pbmc1k"</span><span class="op">)</span></span>
<span><span class="va">adt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/CreateAssayObject.html" class="external-link">CreateAssayObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">raw.matrices</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">data.raw</span><span class="op">[[</span><span class="st">"ADT"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">adt</span></span>
<span></span>
<span><span class="va">nonempty.matrices</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/Read10X.html" class="external-link">Read10X</a></span><span class="op">(</span>data.dir <span class="op">=</span> <span class="st">"../data/filtered_feature_bc_matrix"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 10X data contains more than one type and is being returned as a list containing matrices of each type.</span></span>
<span><span class="va">data.nonartifact</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">nonempty.matrices</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, assay <span class="op">=</span> <span class="st">"RNA"</span>, project <span class="op">=</span> <span class="st">"pbmc1k"</span><span class="op">)</span></span>
<span><span class="va">adt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/CreateAssayObject.html" class="external-link">CreateAssayObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">nonempty.matrices</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">data.nonartifact</span><span class="op">[[</span><span class="st">"ADT"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">adt</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="quality-control">Quality control<a class="anchor" aria-label="anchor" href="#quality-control"></a>
</h4>
<p>The filtered matrix will be further subsetted based on mitochondrial
percentage and total number of features detected:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data.nonartifact</span><span class="op">[[</span><span class="st">"percent.mt"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/PercentageFeatureSet.html" class="external-link">PercentageFeatureSet</a></span><span class="op">(</span><span class="va">data.nonartifact</span>, pattern <span class="op">=</span> <span class="st">"^MT-"</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">data.nonartifact</span>, subset <span class="op">=</span> <span class="va">nFeature_RNA</span> <span class="op">&gt;</span> <span class="fl">200</span> <span class="op">&amp;</span> <span class="va">percent.mt</span> <span class="op">&lt;</span> <span class="fl">5</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="clustering">Clustering<a class="anchor" aria-label="anchor" href="#clustering"></a>
</h4>
<p>The goal of many single cell workflows are to perform clustering to
identify biologically distrinct groups of cells. We will perform
clustering on all QC’ed cells using RNA, ADT, or both and then also
re-cluster the T-cells by themselves.</p>
</div>
<div class="section level4">
<h4 id="analyzing-all-cells">Analyzing all cells<a class="anchor" aria-label="anchor" href="#analyzing-all-cells"></a>
</h4>
<div class="section level5">
<h5 id="rna-only-clustering-of-all-cells">RNA only clustering of all cells<a class="anchor" aria-label="anchor" href="#rna-only-clustering-of-all-cells"></a>
</h5>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">'RNA'</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html" class="external-link">FindVariableFeatures</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindNeighbors.html" class="external-link">FindNeighbors</a></span><span class="op">(</span><span class="va">data</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html" class="external-link">FindClusters</a></span><span class="op">(</span><span class="va">data</span>, resolution <span class="op">=</span> <span class="fl">2</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span><span class="va">data</span>, reduction <span class="op">=</span> <span class="st">'pca'</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, assay <span class="op">=</span> <span class="st">'RNA'</span>, </span>
<span>              reduction.name <span class="op">=</span> <span class="st">'rna.umap'</span>, reduction.key <span class="op">=</span> <span class="st">'rnaUMAP_'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">data</span>, reduction <span class="op">=</span> <span class="st">'rna.umap'</span><span class="op">)</span></span></code></pre></div>
<p><img src="seurat_rmams_tutorial_files/figure-html/plot_rna-1.png" width="700"></p>
</div>
<div class="section level5">
<h5 id="adt-only-clustering-of-all-cells">ADT only clustering of all cells<a class="anchor" aria-label="anchor" href="#adt-only-clustering-of-all-cells"></a>
</h5>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">'ADT'</span></span>
<span><span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/VariableFeatures.html" class="external-link">VariableFeatures</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">data</span><span class="op">[[</span><span class="st">"ADT"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">data</span>, normalization.method <span class="op">=</span> <span class="st">'CLR'</span>, margin <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span>reduction.name <span class="op">=</span> <span class="st">'apca'</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindNeighbors.html" class="external-link">FindNeighbors</a></span><span class="op">(</span><span class="va">data</span>, reduction <span class="op">=</span> <span class="st">"apca"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html" class="external-link">FindClusters</a></span><span class="op">(</span><span class="va">data</span>, graph.name <span class="op">=</span> <span class="st">"ADT_snn"</span>, resolution <span class="op">=</span> <span class="fl">2</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span><span class="va">data</span>, reduction <span class="op">=</span> <span class="st">'apca'</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, </span>
<span>              reduction.name <span class="op">=</span> <span class="st">'adt.umap'</span>, reduction.key <span class="op">=</span> <span class="st">'adtUMAP_'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">data</span>, reduction <span class="op">=</span> <span class="st">'adt.umap'</span><span class="op">)</span></span></code></pre></div>
<p><img src="seurat_rmams_tutorial_files/figure-html/plot_adt-1.png" width="700"></p>
</div>
<div class="section level5">
<h5 id="multimodal-clustering-of-all-cells">Multimodal clustering of all cells<a class="anchor" aria-label="anchor" href="#multimodal-clustering-of-all-cells"></a>
</h5>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindMultiModalNeighbors.html" class="external-link">FindMultiModalNeighbors</a></span><span class="op">(</span></span>
<span>  <span class="va">data</span>, reduction.list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"pca"</span>, <span class="st">"apca"</span><span class="op">)</span>, </span>
<span>  dims.list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span>, modality.weight.name <span class="op">=</span> <span class="st">"RNA.weight"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span><span class="va">data</span>, nn.name <span class="op">=</span> <span class="st">"weighted.nn"</span>, reduction.name <span class="op">=</span> <span class="st">"wnn.umap"</span>, reduction.key <span class="op">=</span> <span class="st">"wnnUMAP_"</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html" class="external-link">FindClusters</a></span><span class="op">(</span><span class="va">data</span>, graph.name <span class="op">=</span> <span class="st">"wsnn"</span>, algorithm <span class="op">=</span> <span class="fl">3</span>, resolution <span class="op">=</span> <span class="fl">2</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">data</span>, reduction <span class="op">=</span> <span class="st">'wnn.umap'</span>, label <span class="op">=</span> <span class="cn">TRUE</span>, repel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="seurat_rmams_tutorial_files/figure-html/plot_multimodal-1.png" width="700"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/FeaturePlot.html" class="external-link">FeaturePlot</a></span><span class="op">(</span><span class="va">data</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"adt_CD3"</span>,<span class="st">"rna_CD3E"</span><span class="op">)</span>, reduction <span class="op">=</span> <span class="st">'wnn.umap'</span><span class="op">)</span></span></code></pre></div>
<p><img src="seurat_rmams_tutorial_files/figure-html/plot_multimodal-2.png" width="700"></p>
</div>
</div>
<div class="section level4">
<h4 id="analysis-of-t-cell-subset">Analysis of T-cell subset<a class="anchor" aria-label="anchor" href="#analysis-of-t-cell-subset"></a>
</h4>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data.tcell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">data</span>, <span class="va">seurat_clusters</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fl">0</span><span class="op">:</span><span class="fl">7</span><span class="op">)</span></span></code></pre></div>
<div class="section level5">
<h5 id="rna-only-clustering-of-t-cells">RNA only clustering of T-cells<a class="anchor" aria-label="anchor" href="#rna-only-clustering-of-t-cells"></a>
</h5>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">data.tcell</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">'RNA'</span></span>
<span><span class="va">data.tcell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">data.tcell</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html" class="external-link">FindVariableFeatures</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span>reduction.name <span class="op">=</span> <span class="st">"pca.tcell"</span><span class="op">)</span></span>
<span><span class="va">data.tcell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindNeighbors.html" class="external-link">FindNeighbors</a></span><span class="op">(</span><span class="va">data.tcell</span>, reduction <span class="op">=</span> <span class="st">"pca.tcell"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">data.tcell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html" class="external-link">FindClusters</a></span><span class="op">(</span><span class="va">data.tcell</span>, resolution <span class="op">=</span> <span class="fl">2</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">data.tcell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span><span class="va">data.tcell</span>, reduction <span class="op">=</span> <span class="st">'pca.tcell'</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, assay <span class="op">=</span> <span class="st">'RNA'</span>, </span>
<span>              reduction.name <span class="op">=</span> <span class="st">'rna.tcell.umap'</span>, reduction.key <span class="op">=</span> <span class="st">'rnaTcellUMAP_'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">data.tcell</span>, reduction <span class="op">=</span> <span class="st">'rna.tcell.umap'</span><span class="op">)</span></span></code></pre></div>
<p><img src="seurat_rmams_tutorial_files/figure-html/plot_rna_tcell-1.png" width="700"></p>
</div>
<div class="section level5">
<h5 id="adt-only-clustering-of-t-cells">ADT only clustering of T-cells<a class="anchor" aria-label="anchor" href="#adt-only-clustering-of-t-cells"></a>
</h5>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">data.tcell</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">'ADT'</span></span>
<span><span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/VariableFeatures.html" class="external-link">VariableFeatures</a></span><span class="op">(</span><span class="va">data.tcell</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">data.tcell</span><span class="op">[[</span><span class="st">"ADT"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">data.tcell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">data.tcell</span>, normalization.method <span class="op">=</span> <span class="st">'CLR'</span>, margin <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span>reduction.name <span class="op">=</span> <span class="st">'apca.tcell'</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">data.tcell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span><span class="va">data.tcell</span>, reduction <span class="op">=</span> <span class="st">'apca.tcell'</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, assay <span class="op">=</span> <span class="st">'ADT'</span>, </span>
<span>              reduction.name <span class="op">=</span> <span class="st">'adt.tcell.umap'</span>, reduction.key <span class="op">=</span> <span class="st">'adtTcellUMAP_'</span><span class="op">)</span></span>
<span><span class="va">data.tcell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindNeighbors.html" class="external-link">FindNeighbors</a></span><span class="op">(</span><span class="va">data.tcell</span>, reduction <span class="op">=</span> <span class="st">"apca.tcell"</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">data.tcell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html" class="external-link">FindClusters</a></span><span class="op">(</span><span class="va">data.tcell</span>, graph.name <span class="op">=</span> <span class="st">"ADT_snn"</span>, resolution <span class="op">=</span> <span class="fl">2</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">data.tcell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html" class="external-link">FindClusters</a></span><span class="op">(</span><span class="va">data.tcell</span>, algorithm <span class="op">=</span> <span class="fl">3</span>, resolution <span class="op">=</span> <span class="fl">2</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">data.tcell</span>, reduction <span class="op">=</span> <span class="st">'adt.tcell.umap'</span><span class="op">)</span></span></code></pre></div>
<p><img src="seurat_rmams_tutorial_files/figure-html/plot_adt_tcell-1.png" width="700"></p>
</div>
<div class="section level5">
<h5 id="multimodal-clustering-of-t-cells">Multimodal clustering of T-cells<a class="anchor" aria-label="anchor" href="#multimodal-clustering-of-t-cells"></a>
</h5>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data.tcell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindMultiModalNeighbors.html" class="external-link">FindMultiModalNeighbors</a></span><span class="op">(</span></span>
<span>  <span class="va">data.tcell</span>, reduction.list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"pca.tcell"</span>, <span class="st">"apca.tcell"</span><span class="op">)</span>,</span>
<span>  dims.list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span>, modality.weight.name <span class="op">=</span> <span class="st">"RNA.weight"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data.tcell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span><span class="va">data.tcell</span>, nn.name <span class="op">=</span> <span class="st">"weighted.nn"</span>, reduction.name <span class="op">=</span> <span class="st">"wnn.tcell.umap"</span>, reduction.key <span class="op">=</span> <span class="st">"wnnTcellUMAP_"</span><span class="op">)</span></span>
<span><span class="va">data.tcell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html" class="external-link">FindClusters</a></span><span class="op">(</span><span class="va">data.tcell</span>, graph.name <span class="op">=</span> <span class="st">"wsnn"</span>, algorithm <span class="op">=</span> <span class="fl">3</span>, resolution <span class="op">=</span> <span class="fl">2</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">data.tcell</span>, reduction <span class="op">=</span> <span class="st">'wnn.tcell.umap'</span>, label <span class="op">=</span> <span class="cn">TRUE</span>, repel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="seurat_rmams_tutorial_files/figure-html/plot_multimodal_tcell-1.png" width="700">
The data matrices stored within a Seurat object will be saved to an
“rds” file. Note that we could extract and save each matrix in a basic
txt or mtx file as well.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">data.raw</span>, <span class="st">"../inst/extdata/pbmc1k_seurat_raw.rds"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">data.nonartifact</span>, <span class="st">"../inst/extdata/pbmc1k_seurat_nonartifact.rds"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">data</span>, <span class="st">"../inst/extdata/pbmc1k_seurat_filtered.rds"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">data.tcell</span>, <span class="st">"../inst/extdata/pbmc1k_seurat_tcell.rds"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level3">
<h3 id="step-2-creating-the-mams-object-from-seurat-object">Step 2: Creating the MAMS object from seurat object<a class="anchor" aria-label="anchor" href="#step-2-creating-the-mams-object-from-seurat-object"></a>
</h3>
<div class="section level4">
<h4 id="loading-data">Loading data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h4>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://single-cell-mams.github.io/rmams/">rmams</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># get the file paths of seurat objects</span></span>
<span></span>
<span><span class="va">file_paths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"../inst/extdata/pbmc1k_seurat_filtered.rds"</span>, <span class="st">"../inst/extdata/pbmc1k_seurat_nonartifact.rds"</span>, <span class="st">"../inst/extdata/pbmc1k_seurat_tcell.rds"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># get the names of these objects </span></span>
<span><span class="va">file_names</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">"\\.rds$"</span>, replacement <span class="op">=</span> <span class="st">""</span>, x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/path.html" class="external-link">basename</a></span><span class="op">(</span><span class="va">file_paths</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># save the seurat objects onto the object_list by reading the object from file path.</span></span>
<span><span class="va">object_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">file_paths</span>, <span class="va">readRDS</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># name the list objects using previously generated file_names.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">object_list</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">file_names</span></span>
<span></span>
<span><span class="co"># define the type of each subset</span></span>
<span><span class="va">observational_subsets</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"filtered"</span>,<span class="st">"nonartifact"</span>,<span class="st">"subset"</span><span class="op">)</span></span>
<span><span class="va">datasetid</span> <span class="op">=</span> <span class="st">"PBMC1K"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="create-mams-object">Create MAMS object<a class="anchor" aria-label="anchor" href="#create-mams-object"></a>
</h4>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">parents</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"yes"</span>,<span class="st">"no"</span>,<span class="st">"no"</span>,<span class="st">"no"</span><span class="op">)</span></span>
<span><span class="va">result</span><span class="op">&lt;-</span> <span class="fu"><a href="../reference/convert_seurat_to_MAMS.html">convert_seurat_to_MAMS</a></span><span class="op">(</span><span class="va">object_list</span>,<span class="va">observational_subsets</span>,<span class="va">datasetid</span>,<span class="va">parents</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Layer 'scale.data' is empty</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="accessing-data">Accessing data<a class="anchor" aria-label="anchor" href="#accessing-data"></a>
</h4>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get the dataset_id from the first fom</span></span>
<span><span class="fu"><a href="../reference/FOM-get-generic.html">fom</a></span><span class="op">(</span><span class="va">result</span>,<span class="st">"fom1"</span>,<span class="st">"dataset_id"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "PBMC1K"</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="step-3-validating-mams-object">Step 3: Validating MAMS object<a class="anchor" aria-label="anchor" href="#step-3-validating-mams-object"></a>
</h3>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/FOM-get-generic.html">fom</a></span><span class="op">(</span><span class="va">result</span>,<span class="st">"fom1"</span>,<span class="st">"dataset_id"</span><span class="op">)</span> <span class="op">&lt;-</span><span class="st">""</span></span>
<span><span class="fu"><a href="../reference/check_MAMS.html">check_MAMS</a></span><span class="op">(</span>mams_object <span class="op">=</span> <span class="va">result</span><span class="op">)</span></span>
<span><span class="co">#&gt; Error: Please use the setter functions to add the missing field information for the following slots: </span></span>
<span><span class="co">#&gt; (Ex. If missing 'filepath' for fom1, then 'fom(mams = mams_object, fom_id = 'fom1', key = 'filepath') &lt;- ...')</span></span>
<span><span class="co">#&gt; :</span></span>
<span><span class="co">#&gt; Warning in check_MAMS(mams_object = result): fom1 is missing the fields:</span></span>
<span><span class="co">#&gt; [1] "dataset_id"</span></span>
<span><span class="co">#&gt; Warning in check_MAMS(mams_object = result):</span></span>
<span><span class="co">#&gt; Warning in check_MAMS(mams_object = result): The following slot is empty: FNG</span></span></code></pre></div>
<div class="section level4">
<h4 id="step-4-filling-the-missing-field">Step 4: Filling the missing field<a class="anchor" aria-label="anchor" href="#step-4-filling-the-missing-field"></a>
</h4>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/FOM-get-generic.html">fom</a></span><span class="op">(</span><span class="va">result</span>,<span class="st">"fom1"</span>,<span class="st">"dataset_id"</span><span class="op">)</span> <span class="op">&lt;-</span><span class="st">"PBMC1K"</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="step-4-saving-mams-object-to-json">Step 4: Saving MAMS object to JSON<a class="anchor" aria-label="anchor" href="#step-4-saving-mams-object-to-json"></a>
</h3>
<div class="section level4">
<h4 id="here-we-show-how-you-can-convert-the-mams-object-to-the-json-format-for-export--mams-and-json-formats-are-interchangeable-">Here we show how you can convert the MAMS object to the JSON format
for export. MAMS and JSON formats are interchangeable.<a class="anchor" aria-label="anchor" href="#here-we-show-how-you-can-convert-the-mams-object-to-the-json-format-for-export--mams-and-json-formats-are-interchangeable-"></a>
</h4>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/write_MAMS_to_output.html">write_MAMS_to_output</a></span><span class="op">(</span>MAMS<span class="op">=</span><span class="va">result</span>, filepath <span class="op">=</span> <span class="st">"../inst/extdata/pbmc1k_rna_adt_seurat_mams.JSON"</span>,format <span class="op">=</span> <span class="st">"JSON"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="step-5-saving-mams-as-yaml">Step 5: Saving MAMS as YAML<a class="anchor" aria-label="anchor" href="#step-5-saving-mams-as-yaml"></a>
</h3>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/write_MAMS_to_output.html">write_MAMS_to_output</a></span><span class="op">(</span>MAMS<span class="op">=</span><span class="va">result</span>, filepath <span class="op">=</span> <span class="st">"../inst/extdata/pbmc1k_rna_adt_seurat_mams.yml"</span>,format <span class="op">=</span> <span class="st">"YAML"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="support-for-seurat-v5-objects">Support for Seurat V5 objects<a class="anchor" aria-label="anchor" href="#support-for-seurat-v5-objects"></a>
</h3>
<div class="section level5">
<h5 id="in-this-example-we-show-how-we-can-extract-counts-from-seurat-v5-object-and-make-a-v3-object-that-can-be-used-with-rmams-pacakge">In this example we show how we can extract counts from Seurat V5
object and make a V3 object that can be used with rmams pacakge<a class="anchor" aria-label="anchor" href="#in-this-example-we-show-how-we-can-extract-counts-from-seurat-v5-object-and-make-a-v3-object-that-can-be-used-with-rmams-pacakge"></a>
</h5>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>Seurat.object.assay.version <span class="op">=</span> <span class="st">"v5"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">pbmc.counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/Read10X.html" class="external-link">Read10X</a></span><span class="op">(</span>data.dir <span class="op">=</span> <span class="st">"../data/filtered_feature_bc_matrix/"</span><span class="op">)</span></span>
<span><span class="va">pbmc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">pbmc.counts</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">pbmc</span><span class="op">[[</span><span class="st">"RNA"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Assay5"</span></span>
<span><span class="co">#&gt; attr(,"package")</span></span>
<span><span class="co">#&gt; [1] "SeuratObject"</span></span>
<span></span>
<span><span class="co"># Extract counts from v5</span></span>
<span><span class="va">v3_counts</span> <span class="op">=</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/Layers.html" class="external-link">LayerData</a></span><span class="op">(</span><span class="va">pbmc</span>, assay<span class="op">=</span><span class="st">"RNA"</span>, layer<span class="op">=</span><span class="st">'counts'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># make v3 object </span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>Seurat.object.assay.version <span class="op">=</span> <span class="st">"v3"</span><span class="op">)</span></span>
<span><span class="va">pbmc_v3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">v3_counts</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">pbmc_v3</span><span class="op">[[</span><span class="st">"RNA"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Assay"</span></span>
<span><span class="co">#&gt; attr(,"package")</span></span>
<span><span class="co">#&gt; [1] "SeuratObject"</span></span>
<span></span>
<span><span class="co"># This pbmc_v3 can now be used with rmams pacakge</span></span></code></pre></div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Irzam Sarfraz, Ming Liu, Salam Alabdullatif, Yusuke Koga, Amulya Shastry, Natasha Gurevich, Junxiang Xu, Tomoki Motegi, Jingwen Xu, Tao Guo, Joshua David Campbell.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
